File: Core/DotNetCore/MediaHatchCore/Media.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace MediaHatchCore
{
    public abstract class Media
    {
		[JsonProperty("rawName")]
        public string RawName { get; set; }

		[JsonProperty("name")]
        public string Name { get; set; }

		public virtual string FullName(string name = null)
		{
			return name ?? Name;
		}

		//[JsonProperty("description")]
        public string Description { get; set; }

		//[JsonProperty("illustrationUrl")]
		public string IllustrationUrl { get; set; }

        public byte[] Illustration { get; set; }

        public List<LinksPackage> LinksPackages { get; set; }

		[JsonProperty("tags")]
		public string[] Tags { get; set; }

		public bool LinksPackageAvailable => LinksPackages?.Count > 0;

        public LinksPackage GetFirstFavoriteLinksPackage(List<string> orderedFavoriteFileHosts)
        {
            for (int i = 0; i < orderedFavoriteFileHosts.Count; i++)
            {
                LinksPackage linksPackage = LinksPackages.FirstOrDefault(lp => lp.FileHost.Contains(orderedFavoriteFileHosts[i]));
                if (linksPackage != null)
                    return linksPackage;
            }
            if (LinksPackages.Count > 0)
                return LinksPackages[0];
            else
                return null;
        }
    }
}

File: Core/media-hatch-core/src/media.ts
import { LinksPackage } from './links-package';
import { BaseElement } from './base-element';
import { JsonObject, JsonProperty } from 'json2typescript';

@JsonObject()
export abstract class Media extends BaseElement {
	constructor(type: string) {
		super(type);
	}

	@JsonProperty('rawName')
	public rawName: string = undefined;

	@JsonProperty('name')
	public name: string = undefined;

	public get fullName(): string {
		return this.name;
	}

	@JsonProperty('description')
	public description: string = undefined;
	public illustration;

	@JsonProperty('illustrationUrl')
	public illustrationUrl: string = undefined;

	public linksPackages: LinksPackage[];

	@JsonProperty('tags', [String])
	public tags = new Array<string>();

	@JsonProperty('size')
	public size: string = undefined;

	public get linksPackageAvailable(): boolean {
		return this.linksPackages && this.linksPackages.length > 0;
	}
	public getFirstFavoriteLinksPackage(orderedFavoriteFileHosts: string[]): LinksPackage {
		if (this.linksPackageAvailable) {
			for (let fileHost of orderedFavoriteFileHosts) {
				for (let linksPackage of this.linksPackages) {
					if (linksPackage.fileHost.startsWith(fileHost)) {
						return linksPackage;
					}
				}
			}
			return this.linksPackages[0];
		} else {
			return undefined;
		}
	}
}
